scenarios/throughput_stress.go

```diff
@@ -43,14 +43,16 @@ func (t *tpsExecutor) Run(ctx context.Context, info loadgen.ScenarioInfo) error
 	var deniedErr *serviceerror.PermissionDenied
 	var alreadyErr *serviceerror.AlreadyExists
 
-	if errors.As(err, &deniedErr) {
+	if errors.As(err, &alreadyErr) {
+		info.Logger.Infof("Search Attribute %s already exists", ThroughputStressScenarioIdSearchAttribute)
+	} else if err != nil {
 		info.Logger.Warnf("Failed to add Search Attribute %s: %v", ThroughputStressScenarioIdSearchAttribute, err)
-	} else if !errors.As(err, &alreadyErr) {
-		info.Logger.Info("Search Attribute %s already exists", ThroughputStressScenarioIdSearchAttribute)
 
-		return err
+		if !errors.As(err, &deniedErr) {
+			return err
+		}
```

Comment:
We have to swallow permission denied, otherwise users hitting Cloud will never be able to use this scenario. The search attribute creation API call returns permission denied regardless of whether the search attribute exists on Cloud.

Reply:
Ah, I see, I misread at first. We probably need a general "ensure search attributes" utility at some point, but this LGTM.